lib/virt_autotest/utils.pm

```diff
@@ -42,39 +42,58 @@ sub trim {
     return $text;
 }
 
-sub restart_libvirtd {
-    if (is_sle('<12')) {
-        assert_script_run('rclibvirtd restart', 180);
-    }
-    elsif (is_alp) {
-        my $_libvirtd_pid = script_output(q@ps -ef |grep [l]ibvirtd | gawk '{print $2;}'@);
-        my $_libvirtd_cmd = script_output("ps -o command $_libvirtd_pid | tail -1");
-        assert_script_run("kill -9 $_libvirtd_pid");
-        assert_script_run("$_libvirtd_cmd");
-    }
-    else {
-        systemctl("restart libvirtd", timeout => 180);
-    }
-    record_info("Libvirtd has been restarted!");
-    save_screenshot;
-}
-
 #return 1 if test is expected to run on XEN hypervisor
 sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
+#Setup modular libvrit daemons as the default in SLE15SP6, refer to PED-4804 to get more details
+sub is_modular_libvirtd {
+    return 0 if is_sle(">=15-sp6");
+}
+
+# Usage: check_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
+# to specify daemons which will be checked: check_modular_libvirt_daemons(qemu storage ...)
+# to check all required modular daemons without any daemons passed
+sub check_modular_libvirt_daemons {
+    my @daemons;
+
+    if (@_ == 0) {
+        @daemons = qw(network nodedev nwfilter secret storage proxy log lock);
+        if (is_xen_host) {
+            push @daemons, 'xen';
+        } else {
+            push @daemons, 'qemu';
+        }
+    } else {
+        @daemons = @_;
+    }
+
+    foreach my $daemon (@daemons) {
+        systemctl("status virt${daemon}d.service");
+        if ($daemon == 'lock') {
```

Comment:
It seems that there are also virtproxyd and virtlogd. Do you want to cover them ?

Reply:
good point! there should be cover them together.

Reply:
Short update,  confirm that virtlogd and virtlockd without virtproxyd-ro.socket and virtlockd-admin.socket. 
meanwhile, virtproxyd have not this problem:
# systemctl status virtproxyd
virtproxyd-admin.socket  virtproxyd.service       virtproxyd-tcp.socket    
virtproxyd-ro.socket     virtproxyd.socket        virtproxyd-tls.socket

FYI
# systemctl status virtlogd
virtlogd-admin.socket  virtlogd.service       virtlogd.socket
# systemctl status virtlockd
virtlockd-admin.socket  virtlockd.service       virtlockd.socket